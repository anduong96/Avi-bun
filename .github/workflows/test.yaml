name: Tests

on:
  push:
    branches: [main]
  pull_request:
    branches: [main]

permissions:
  checks: write
  contents: write

jobs:
  run-tests:
    name: Run Tests
    runs-on: ubuntu-latest
    env:
      DOPPLER_TOKEN: ${{ secrets.DOPPLER_TOKEN }}
      ENV: test

    steps:
      - name: Check out Git repository
        uses: actions/checkout@v3

      - uses: oven-sh/setup-bun@v1
        with:
          bun-version: latest

      - name: 'Install dependencies'
        run: bun install --silent

      - name: Generate prisma schema
        run: bun run prisma:generate

      - name: Run tests
        run: bun run test:coverage
