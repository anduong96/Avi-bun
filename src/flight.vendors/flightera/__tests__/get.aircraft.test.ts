import { describe, expect, test } from 'bun:test';

import { Flightera } from '..';

describe('Flightera', () => {
  test('Aircraft: N725AN', async () => {
    const aircraftTailNumber = 'N725AN';
    const aircraft = await Flightera.getAircraftFromCrawl(aircraftTailNumber);
    expect(aircraft).toMatchSnapshot();
  });
});
