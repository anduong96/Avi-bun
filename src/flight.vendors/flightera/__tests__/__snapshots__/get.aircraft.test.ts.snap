// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Flightera Aircraft: N725AN 1`] = `
{
  "airlineIata": "AA",
  "description": "The Boeing 777-300ER (Extended Range), also known as the B77W, is a long-haul, twin-engine airliner. It boasts an impressive range and a large passenger capacity, typically around 396 passengers. Known for its fuel efficiency, spacious cabin, and high-tech features, it's often used for international and transcontinental flights.",
  "firstFlight": 2013-07-01T00:00:00.000Z,
  "icao": "A9B62C",
  "image": "https://www.flightera.net/staticfiles/acftpic_131875.jpg",
  "model": "B77W",
  "seatsConfiguration": {
    "business": 52,
    "economy": 216,
    "first": 8,
  },
}
`;
